src/power_grid_model/core/buffer_handling.py

```diff
@@ -127,121 +211,197 @@ def _get_sparse_buffer_properties(data: SparseBatchArray) -> BufferProperties:
     Returns:
         the properties of the dataset component.
     """
-    is_sparse = True
+    is_sparse_property = True
 
     contents = data["data"]
     indptr = data["indptr"]
 
-    if contents.ndim != 1:
-        raise ValueError(f"Data array can only be 1D. {VALIDATOR_MSG}")
+    if not isinstance(indptr, np.ndarray):
+        raise TypeError("indptr must be of type IndexPointer [np.ndarray]")
+
+    ndim = 1
+    columns: list[AttributeType] | None = None
+    if isinstance(contents, np.ndarray):
+        shape: tuple[int, ...] = contents.shape
+    elif not contents:
+        raise ValueError("Empty columnar buffer is ambiguous. {VALIDATOR_MSG}")
```

Comment:
forgotten
```suggestion
        raise ValueError(f"Empty columnar buffer is ambiguous. {VALIDATOR_MSG}")
```

please also go over all other instances